tests/integration/test_cli_dedupe_hash_undo_interactive_daemon.py

```
@@ -0,0 +1,1001 @@
+        call_kwargs = mock_cfg_cls.call_args.kwargs
+        assert call_kwargs["debounce_seconds"] == pytest.approx(2.5)
+
+    def test_watch_event_src_path_fallback(self, tmp_path: Path) -> None:
+        """Events without .path attribute fall back to .src_path."""
+        mock_event = MagicMock(spec=[])  # no attributes via spec
+        mock_event.event_type = "modified"
+        mock_event.src_path = "/fallback/file.txt"
+
+        call_count = 0
+
+        def _get_events(**kwargs):
```

Ruff (0.15.9)

[warning] Missing return type annotation for private function `_get_events`

(ANN202)